core/src/main/kotlin/org/eln2/sim/thermal/main.kt

```diff
@@ -0,0 +1,146 @@
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.*
+import kotlin.math.*
+
+class ThermalInterface{
+
+    var mutex = Mutex()
+    var R = 1f //heat resistance at the interface
+    var heatTranfer = 0f //heat transfer during a simulation tick through this interface
+
+    var lastTickComputed = 0 //number of the last tick where the heat transfer was computed. Tick number switch between 0 and 1. This is here to prevent computing twice the heat trasnfered when not needed
```

Comment:
If the domain is finite _and_ size two, do you think a boolean would work better here?

Reply:
It should work the same, but I find it easier to read that the tickNumber is different. It might also help with debugging since you can actually keep track of the ticknumber if you want to, instead of being restricted to 0 and 1. 

That being said, I don't really have a strong opinion on that

Reply:
I guess I don't either, beyond my "strongest typing" smell from other languages. Besides, managed language again; there's probably not a _whole_ lot of an optimization to be had here (if this were C, I'd definitely have dropped it into binary and/xor already :P). Whatever's clear seems good :)